### ISSUE TITLE
Redirect URL misconfiguration

### ISSUE THREAD

**Issue by brandonhilkert** [murbanski/omniauth-microsoft-office365#1]
Not sure if you can help me here, but I'm looking to use this get access to users' hotmail/outlook email. I have an app where we integrate about 15 other providers right now and all work as expected.

With this strategy, I consistently get the following error:

``` json
{"error":"invalid_grant","error_description":"AADSTS50011: The reply address 'http://localhost:3000/auth/outlook/callback?code=M25695f93-01a4-8910-609c-5e9c1be767d8&state=f6a085391887b9f98b7808bf0ced677fa125a6f17acbfcb6' does not match the reply addresses configured for the application: 7bc8f03e-1b89-4033-b048-84f6fc6e43f4.\r\nTrace ID: 41f57ca9-410a-447a-86f2-48a940a647d7\r\nCorrelation ID: 58d42ed2-b699-4477-b4a1-22307c879c18\r\nTimestamp: 2016-04-13 14:24:25Z","error_codes":[50011],"timestamp":"2016-04-13 14:24:25Z","trace_id":"41f57ca9-410a-447a-86f2-48a940a647d7","correlation_id":"58d42ed2-b699-4477-b4a1-22307c879c18"}
```

And I've setup the redirect URL like so:

<img width="528" alt="screen shot 2016-04-13 at 10 41 38 am" src="https://cloud.githubusercontent.com/assets/744212/14497280/5672ba34-0164-11e6-9df6-de2943bdf50a.png">

``` ruby
module OmniAuth::Strategies
  class Outlook < MicrosoftOffice365
    def name
      :outlook
    end
  end
end

Rails.application.config.middleware.use OmniAuth::Builder do
  provider :outlook, ENV['OUTLOOK_CLIENT_ID'], ENV['OUTLOOK_CLIENT_SECRET'],
    {
      authorize_params: {
        scope: "openid email profile https://outlook.office.com/mail.read"
      }
    }
end
```

Just curious if you've seen this at all. I went through [their tutorial](https://dev.outlook.com/restapi/tutorial/ruby) and it worked as expected, so my assumption was it was something in the layers of the strategy/omniauth, but the errors don't seem to make sense because that is the URL configured.

Thanks!

**Comment by brandonhilkert** [comment]
I stepped through the Omniauth-oauth2 layer and found that if on the request phase, it includes query strings (rightfully so), and then on the access token phase, it includes the `code` and `state` in the URL and finds that it didn't match the URL in the request phase. It feels like something is wrong under the covers here.

I'm passing a dynamic query string along with the request because I want it to be present during the callback, but it seems like Microsoft's configuration won't allow you to include a query string (it's dynamic anyway so I couldn't make this work) so I can't even test to see if that's the problem.

**Comment by murbanski** [comment]
@brandonhilkert I just released version `0.0.5` of this gem with your patch applied if you are interested.

**Comment by brandonhilkert** [comment]
Awesome! Thanks for the quick turnaround.

On Wednesday, April 13, 2016, Marcin Urbański notifications@github.com
wrote:

> @brandonhilkert https://github.com/brandonhilkert I just released
> version 0.0.5 of this gem with your patch applied if you are interested.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/murbanski/omniauth-microsoft-office365/issues/1#issuecomment-209518667

## 

---

http://brandonhilkert.com
